ArduCopter/mode_circle.cpp

```diff
@@ -96,6 +105,16 @@ void Copter::ModeCircle::run()
     pos_control->update_z_controller();
 }
 
+float Copter::ModeCircle::get_desired_radial_velocity(float norm_pitch_input)
+{
+    return WPNAV_LOITER_SPEED * norm_pitch_input;
+}
+
+float Copter::ModeCircle::get_target_radius(float current_radius, float radial_velocity, float dt)
```

Comment:
This can go negative, which is fun.

Reply:
That was intentional, for allowing the UAV to shrink the radius of the circle. Could this be a problem?

Reply:
Ah, yes, I fixed this.